src/Core/Managed/Shared/Utils.cs

```diff
@@ -74,6 +74,20 @@ public static TimeSpan ValidateDuration(string value)
             return interval;
         }
 
+        /// <summary>
+        /// Returns min DateTimeOffset value if not a valid DateTimeOffset.
+        /// </summary>
+        public static DateTimeOffset ValidateDateTimeOffset(string value)
+        {
+            DateTimeOffset timestamp;
+            if (!DateTimeOffset.TryParse(value, CultureInfo.InvariantCulture, DateTimeStyles.AssumeLocal, out timestamp))
```

Comment:
Probably a question for @MeniZalzman This was coped directly from RequestTelemetry, I made the assumption that validating offsets in AvailabilityTelemetry would need to be the same.